src/autocomplete/RoomProvider.js

```diff
@@ -56,15 +56,31 @@ export default class RoomProvider extends AutocompleteProvider {
         const {command, range} = this.getCurrentCommand(query, selection, force);
         if (command) {
             // the only reason we need to do this is because Fuse only matches on properties
-            this.matcher.setObjects(client.getRooms().filter(
+            let matcherObjects = client.getRooms().filter(
                 (room) => !!room && !!getDisplayAliasForRoom(room),
             ).map((room) => {
                 return {
                     room: room,
                     name: room.name,
                     displayedAlias: getDisplayAliasForRoom(room),
                 };
-            }));
+            });
+
+            // Filter out any matches where the user will have also autocompleted new rooms
+            matcherObjects = matcherObjects.filter((r) => {
+                const tombstone = r.room.currentState.getStateEvents("m.room.tombstone", "");
+                if (tombstone && tombstone.getContent() && tombstone.getContent()['replacement_room']) {
+                    console.log(r.displayedAlias);
+                    const hasReplacementRoom = matcherObjects.some(
```

Comment:
This inner loop could slow down with big accounts I suppose? Wouldn't it be enough to exclude any rooms that have a tombstone event?

Reply:
There's the possibility that, for some reason, the user hasn't joined the upgraded room and the alias is still pointing at the old room. Out of abundance of caution, we might as well verify the room. On my massive account of ~2000 rooms, there's no noticeable performance impact of adding this check (in fact, it feels faster...)